pkg/operator/sync.go

```diff
@@ -97,6 +97,50 @@ func (optr *Operator) syncAll(syncFuncs []syncFunc) error {
 	return syncErr.err
 }
 
+// Return true if cloud config is required on a platform.
+func isCloudConfigRequired(infra *configv1.Infrastructure) bool {
+	if infra.Spec.CloudConfig.Name != "" {
```

Comment:
@abhinavdahiya wondering if we really need this return? Isn't supported platform check and return is enough?

Reply:
it's just useful in most cases, since if spec is set, we better have a cloud-config regardless of if the platform really needs it.

Reply:
got it, thanks!